### DISCUSSION TITLE
Multiple model fields using same currency field does not update correctly when either is updated

### DISCUSSION THREAD

**Discussion by Dambre** [django-money/django-money#765] [Q&A]
I have this implementation:
```
base_reward = MoneyField(currency_field_name="currency", max_digits=14, decimal_places=2, default=0)
bonus_reward = MoneyField(currency_field_name="currency", max_digits=14, decimal_places=2, default=0)
service_rate = MoneyField(currency_field_name="currency", max_digits=14, decimal_places=2, default=0)
```
In django admin only the last field in order of form fields updates the currency. How to handle this so I could use single db column for many fields. Can I have it as a separate field on admin?

**Comment by benjaoming** [comment]
@Dambre is it correctly understood that you want to use the SAME database column to choose the currency for ALL money fields?

**Reply by Dambre** [reply]
yes, exactly, looks like MoneyField allows this option.

**Reply by benjaoming** [reply]
Hmm, I'm not sure what the expected behavior would be here from a UI perspective. MoneyField renders a widget for each field with amount+currency.

What it sounds like you want is:

amount1
amount2
amount3
currency

I think your best bet is to accept having 3 different currency fields and then copy from currency1 to currency2 and currency3 when you save the Model.

amount1, currency1
amount2, currency2
amount2, currency3

```python
class MyModel(...):
    # ...
    def save(self, *args, **kwargs):
        self.bonus_reward_currency = self.base_reward_currency
        self.service_rate_currency = self.base_reward_currency
        super().save(*args, **kwargs)
```

I know that this could look like you're saving redundant currency values, but think about pricing in general: It's very very full of such redundant currency information, for each and every money column and each and every row of data. If you think this is too much redundant/implicit data, then you might be better off not choosing MoneyField and django-money.

**Reply by Dambre** [reply]
The manual handling of multiple fields that must use the same currency feels unintuitive. Considering that this library already utilizes multiple fields to create a Money instance, it might be good to have an option to store a single value for multiple fields, which always have to be the same. This thought occurred to me when I discovered `models.CurrencyField`, which suggested declaring the same currency field name for all relevant fields and use currency = CurrencyField() to define that value.
If incorporating this feature isn't in the library's future plans, and you're not considering a pull request, I plan to implement a custom solution on my side and discontinue using this library.

**Reply by benjaoming** [reply]
> The manual handling of multiple fields that must use the same currency feels unintuitive.

You're the first that I hear this from.

You can also contribute to improve the library to your needs, that's why it's open source.

> I plan to implement a custom solution on my side and discontinue using this library.

Okay, good luck!
